### PR TITLE
Feature/dsnd 1957

### PR DESCRIPTION
**Pull request by lbedrosyanCH**

### CODE REVIEW

File: src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java [hunk 1]
```diff
@@ -683,83 +602,65 @@ public void aPSCExistsFor(String companyNumber) throws JsonProcessingException {
...
         Individual expected = objectMapper.readValue(data, Individual.class);
-
         Individual actual = CucumberContext.CONTEXT.get("getResponseBody");
 
         assertThat(actual.getName()).isEqualTo(expected.getName());
         assertThat(actual.getCountryOfResidence()).isEqualTo(expected.getCountryOfResidence());
         assertThat(actual.getNaturesOfControl()).isEqualTo(expected.getNaturesOfControl());
     }
 
-    @After
```
Comment (JAndrewCH) [inline src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java:722]: I think we want to keep this. The tests will run without it but it is good to make sure that you don't end up with containers that aren't cleaned up properly

File: src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java [hunk 2]
```diff
@@ -49,425 +42,200 @@ public class CompanyPscTransformer {
...
-        } else {
-            logger.error("Skipped transforming pscDoc to individualBeneficialOwner");
-            throw new ResourceNotFoundException(HttpStatus.NOT_FOUND,"PscDocument not found");
-        }
-
+            PscDocument pscDocument, Boolean registerView) {
+        logger.info("Attempting to transform pscDocument to IndividualBeneficialOwner",
+                DataMapHolder.getLogMap());
+        IndividualBeneficialOwner individualBo = new IndividualBeneficialOwner();
+        individualBo.setEtag(pscDocument.getData().getEtag());
```
Comment (JAndrewCH) [inline src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java:79]: I like that we are removing the unneeded null checks. The fields where we call document.getData().getX() need to be inside a check that data is not null though or we can get a null pointer exception

File: src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java [hunk 3]
```diff
@@ -49,425 +42,200 @@ public class CompanyPscTransformer {
...
-            return superSecureBeneficialOwner;
-        } else {
-            logger.error("Skipped transforming pscDoc to SuperSecureBeneficialOwner");
-            throw new ResourceNotFoundException(HttpStatus.NOT_FOUND,"PscDocument not found");
-        }
+    public CorporateEntityBeneficialOwner transformPscDocToCorporateEntityBeneficialOwner(
+            PscDocument pscDocument) {
+        logger.info("Attempting to transform pscDocument to CorporateEntityBeneficialOwner",
+                DataMapHolder.getLogMap());
+        CorporateEntityBeneficialOwner corporateEntityBo = new CorporateEntityBeneficialOwner();
```
Comment (JAndrewCH) [inline src/main/java/uk/gov/companieshouse/pscdataapi/transform/CompanyPscTransformer.java:127]: Same change needed for a null check on data for all of these

**Review by JAndrewCH — CHANGES_REQUESTED** [review]

File: src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java (hunk 1 above)
Comment (lbedrosyanCH) [reply src/itest/java/uk/gov/companieshouse/pscdataapi/steps/PscDataSteps.java:722]: This was moved the beggining of the file, to line 85, right after the "@Before". I think it makes sense to have it there

**Review by JAndrewCH — APPROVED** [review]

**Pull request merged**